app/src/interfaces/presentation-links/presentation-links.vue

```diff
@@ -28,35 +32,62 @@ type Link = {
 	url?: string;
 };
 
-export default defineComponent({
-	props: {
-		links: {
-			type: Array as PropType<Link[]>,
-			default: null,
-		},
-	},
-	setup(props) {
-		const values = inject('values', ref<Record<string, any>>({}));
-
-		const linksParsed = computed(() => {
-			return props.links.map((link) => {
-				const parsedLink = omit<Record<string, any>>(link, ['url']);
-				const linkValue = render(link.url ?? '', values.value);
-
-				if (linkValue.startsWith('/')) {
-					parsedLink.to = linkValue;
-				} else {
-					parsedLink.href = linkValue;
-				}
-
-				return parsedLink;
-			});
-		});
-
-		return {
-			linksParsed,
-		};
-	},
+type Props = {
+	links: Link[];
+	collection: string;
+	primaryKey: string;
+};
+
+const props = withDefaults(defineProps<Props>(), {
+	links: () => [],
+});
+
+const values = inject('values', ref<Record<string, any>>({}));
+
+const { collection, primaryKey } = toRefs(props);
+
+const query = computed(() => {
+	const fields = new Set();
+
+	props.links.forEach((link) => {
+		getFieldsFromTemplate(link.url ?? '').forEach((field) => fields.add(field));
+	});
+
+	return {
+		fields: Array.from(fields),
+	} as Query;
+});
+
+const { item } = useItem(collection, primaryKey, query);
+const { fields } = useCollection(collection);
+
+const RELATIONAL_TYPES = ['file', 'files', 'm2o', 'o2m', 'm2m', 'm2a', 'presentation', 'translations', 'group'];
```

Comment:
This should use an existing `const`, or make this a shared `const` if it doesn't exist yet